widget/entry.go

```diff
@@ -40,6 +40,8 @@ type Entry struct {
 	TextStyle   fyne.TextStyle
 	PlaceHolder string
 	OnChanged   func(string) `json:"-"`
+	// Since: 2.0.0
+	OnSubmitted func(string) `json:"-"`
```

Comment:
Should this perhaps be `OnSubmit`? While it doesn’t conform as well with `OnChanged`, it reads a bit better in my opinion.

Reply:
Yes I thought about that too, `widget.Form` also has `OnSubmit`. There was a long conversation in Slack about this (starting with @okratitan's message https://gophers.slack.com/archives/CK5U4BU87/p1602620036368100 and ending with @andydotxyz's https://gophers.slack.com/archives/CK5U4BU87/p1602664647449500)

Reply:
This was indeed discussed. There were various opinions but we did come to an agreement that was documented in the naming docs on the wiki:

> Callback functions that can be set by external developers should be prefixed with “On...” and use the past tense, for example “OnTapped” or “OnChanged”

https://github.com/fyne-io/fyne/wiki/Contributing#Naming